assets/javascripts/cookies.js

```diff
@@ -1,43 +1,110 @@
-function init() {
+function setCookie(name, value, days) {
+    let expires = '';
+    if (days) {
+        const date = new Date();
+        date.setTime(date.getTime() + (days * 24 * 60 * 60 * 1000));
+        expires = '; expires=' + date.toUTCString();
+    }
+    document.cookie = name + '=' + (value || '') + expires + '; path=/';
+}
+
+function updateCookieAnalyticsSettings(analytics) {
+    let cookieSettings = getCookie('cookie_settings');
+    if (!cookieSettings) {
+        cookieSettings = { essential: true, analytics: false };
+    } else {
+        cookieSettings = JSON.parse(cookieSettings);
+    }
+    cookieSettings.analytics = analytics;
+    setCookie('cookie_settings', JSON.stringify(cookieSettings), 365);
+}
 
-const acceptButton = document.querySelector('.js-cookies-button-accept')
-const rejectButton = document.querySelector('.js-cookies-button-reject')
+function getCookie(name) {
+    const value = `; ${document.cookie}`;
+    const parts = value.split(`; ${name}=`);
+    if (parts.length === 2) {return parts.pop().split(';').shift();}
+}
 
-const acceptedBanner = document.querySelector('.js-cookies-accepted')
-const rejectedBanner = document.querySelector('.js-cookies-rejected')
-const questionBanner = document.querySelector('.js-question-banner')
-const cookieBanner = document.querySelector('.js-cookies-banner')
+function checkCookiePreference() {
+    const cookieValue = document.cookie.split('; ').find(row => row.startsWith('cookie_preferences='))
+        ?.split('=')[1];
 
-function showBanner(banner) {
-    questionBanner.setAttribute('hidden', 'hidden')
-    banner.removeAttribute('hidden')
+    if (cookieValue === 'true') {
+        const cookieBanner = document.querySelector('.js-cookies-banner');
+        cookieBanner.setAttribute('hidden', 'hidden');
+    }
+}
 
-    // Shift focus to the banner
-    banner.setAttribute('tabindex', '-1')
-    banner.focus()
+function showNotificationBanner() {
+    const notificationBanner = document.getElementById('notification-banner');
+    notificationBanner.hidden = false;
+    notificationBanner.setAttribute('tabindex', '-1');
+    notificationBanner.focus();
 
-    banner.addEventListener('blur', function () {
-        banner.removeAttribute('tabindex')
-    })
+    notificationBanner.addEventListener('blur', function() {
+        notificationBanner.removeAttribute('tabindex');
+    });
 }
 
-acceptButton.addEventListener('click', function (event) {
-    showBanner(acceptedBanner)
-    event.preventDefault()
-})
+function init() {
+
+    const acceptButton = document.querySelector('.js-cookies-button-accept');
```

Comment:
Could be nice to reference actually where some of this clientside code comes from. Add a comment linking to [govuk-frontend recommended implementation](https://github.com/alphagov/govuk-frontend/blob/main/packages/govuk-frontend-review/src/views/full-page-examples/cookie-banner-client-side/index.njk)